crenel_caves.logic

```diff
@@ -1,17 +1,124 @@
-area CrenelCaves:
-  room BlocksChest
-  room BlocksPuzzle
-  room BlocksBomb
-  room FloodBlock
-  room FountHP
-  room FountRupee
-  room GraybladeEntrance
-  room Guru
-  room HelmaHall
-  room MushroomGap
-  room RockPush
-  room ScrubBomb
-  room ScrubGrip
-  room ScrubHint
-  room SwitchBridge
-  room PotCircle
\ No newline at end of file
+area CrenelCaves:
+  room BlocksChest:
+    node Main:
+      door Staircase
+      door Doorway
+      enemy RedChus
+      chest BlocksChest
+  room BlocksPuzzle:
+    node BlocksPuzzle:
+      door Ladder
+      door BlocksPuzzleDoor
+      enemy Helmasaurs
+    node MushroomGapSouth:
+      door MushroomGapDoor
+      enemy Helmasaur
+    node MushroomGapNorth:
+      door Staircase
+      enemy Helmasaur
+    MushroomGapNorth <-> MushroomGapSouth: (item GustJar | item Cape | item LightArrows | item Cane)
+  room BlocksBomb:
+    node PreBombBlocks:
+      door PreBombBlocks
+      enemy Chus
+    node PostBombBlocks:
+      door PostBombBlocks
+      enemy SilverChu
+    #These blocks respawn when leaving, and it seems to be impossible to go backwards through this room?
+    PreBombBlocks -> PostBombBlocks: item BombBag
+  room FloodBlock: #This is the room w/ 2 chests and a HP on the very left side of crenel right?
+    node FloodBlockEntrance:
+      door FloodBlock
+    node FloodBlockChests:
+      chest LeftFloodBlockChest
+      chest RightFloodBlockChest
+    #These are also respawning bomb-blocks
+    FloodBlockEntrance -> FloodBlockChests: (item BombBag | item Cape)
+    node FloodBlockHP:
+      chest HearPiece
+    #Should grabbing the HP from the water be in logic? Don't think it currently is
+    FloodBlockEntrance -> FloodBlockHP: (item BombBag | item Cape | item Flippers)
+  room FountHP:
+    node Main:
+      door Doorway
+      chest HeartPiece
+  room FountRupee:
+    node Main:
+      door Doorway
+      chest LeftRupee
+      chest CenterRupee
+      chest RightRupee
+  room GraybladeEntrance:
+    node Main:
+      door Entrance
+      exit DojoExit
+    node Switches:
+      flag Switches
+    #Completely guessing here on the actual item names/syntax
+    Main -> Switches: item CanSplit2
+    node SwitchDoor:
+      entrance SwitchDoor

```

Comment:
And just because they are split, doesn't mean we shouldn't still use the same name for the `entrance`/`exit` pair here
```suggestion
      entrance North
```